### DISCUSSION TITLE
Could this be used for multi-handshake authentication protocols?

### DISCUSSION THREAD

**Discussion by franky47** [nextauthjs/next-auth#6747] [Help]
I'm thinking about zero-knowledge password authenticated key exchange protocols such as Secure Remote Password (SRP) or OPAQUE, which require two HTTP requests for signup, and two for login (plus additional client-side cryptographic code around the requests, but which is out of the scope of this question).

Are Providers able to define custom endpoints to implement these protocols?

**Comment by nikgraf** [comment]
@franky47 did you find anything out? I was wondering the same here: https://github.com/nextauthjs/next-auth/discussions/7987

**Reply by franky47** [reply]
I didn't give it much further thought to be honest. I ended up using OPAQUE, but in a completely different tech stack (Fastify backend-end).

**Reply by nikgraf** [reply]
nice, which OPAQUE implementation did you use?

**Reply by franky47** [reply]
The Rust implementation maintained by Facebook, packaged as WASM in two packages (as the client code doesn't usually need the server runtime counterpart).

Available here: https://github.com/47ng/opaque

**Reply by nikgraf** [reply]
Nice!
